saleor/graphql/core/utils/__init__.py

```diff
@@ -30,12 +34,17 @@ def validate_image_file(file, field_name):
         raise ValidationError({field_name: "Invalid file type"})
 
 
-def from_global_id_strict_type(info, global_id, only_type, field="id"):
+def from_global_id_strict_type(
+    global_id: str, only_type: Union[ObjectType, str], field: str = "id"
+) -> str:
     """Resolve a node global id with a strict given type required."""
-    _type, _id = graphene.Node.from_global_id(global_id)
-    graphene_type = info.schema.get_type(_type).graphene_type
-    if graphene_type != only_type:
-        raise ValidationError({field: "Couldn't resolve to a node: %s" % global_id})
+    try:
+        _type, _id = graphene.Node.from_global_id(global_id)
+    except (binascii.Error, UnicodeDecodeError) as exc:
+        raise ValidationError({field: f"Invalid base64 global ID"}) from exc
```

Comment:
Since these errors might be returned to API users, I think it would be enough to tell them that the API failed to find a node of a given ID. Revealing the base64 decoding failed sounds like a low-level implementation detail.